guides/micronaut-oracle-cloud-streaming/chess-game/groovy/src/main/groovy/example/micronaut/chess/GameController.groovy

```diff
@@ -0,0 +1,81 @@
+package example.micronaut.chess
+
+import example.micronaut.chess.dto.GameDTO
+import example.micronaut.chess.dto.GameStateDTO
+import groovy.transform.CompileStatic
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.PathVariable
+import io.micronaut.http.annotation.Post
+import io.micronaut.scheduling.TaskExecutors
+import io.micronaut.scheduling.annotation.ExecuteOn
+import io.reactivex.Single
+
+import static io.micronaut.http.MediaType.APPLICATION_FORM_URLENCODED
+import static io.micronaut.http.MediaType.TEXT_PLAIN
+
+/**
+ * Endpoints called by the front end using Ajax.
+ */
+@Controller('/game')
+@ExecuteOn(TaskExecutors.IO) // <1>
+@CompileStatic
+class GameController {
+
+    private final GameReporter gameReporter
+
+    GameController(GameReporter gameReporter) {
+        this.gameReporter = gameReporter
+    }
+
+    /**
+     * Start a new game.
+     *
+     * @param b black's name
+     * @param w white's name
+     * @return the id of the game
+     */
+    @Post(value = '/start', consumes = APPLICATION_FORM_URLENCODED, produces = TEXT_PLAIN) // <2>
+    Single<String> start(String b, String w) { // <3>
+        GameDTO game = new GameDTO(UUID.randomUUID().toString(), b, w) // <4>
+        gameReporter.game(game.id, game).map(gameDTO -> game.id) // <5>
+    }
+
+    /**
+     * Record a move.
+     *
+     * @param gameId the <code>Game</code> id
+     * @param player b or w
+     * @param move the current move
+     * @param fen the current FEN string
+     * @param pgn the current PGN string
+     */
+    @Post(value = '/move/{gameId}', consumes = APPLICATION_FORM_URLENCODED) // <6>
+    void move(@PathVariable String gameId, String player, String move, String fen, String pgn) {
```

Comment:
I will add a carriage return between the methods parameters.

Reply:
Should we add `@Status(HttpStatus.OK)` to clearly explain what is the status code returned? Or should we say it in the copy?